bundles/framework/geometrycutter/instance.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @class Oskari.mapframework.ui.module.common.geometrycutter.GeometryCutterBundleInstance
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.geometrycutter.GeometryCutterBundleInstance', function () {
+    this._editsInProgress = {};
+}, {
+    __name: 'GeometryCutterBundleInstance',
+    __idPrefix: 'geometryCutter-',
+    eventHandlers: {
+        'DrawingEvent': function (event) {
+            var me = this;
+            var drawId = event.getId();
+            if (drawId.substr(0, this.__idPrefix.length) !== this.__idPrefix || !event.getIsFinished()) {
+                return;
+            }
+            var editState = this._editsInProgress[drawId];
+            if (!editState || !editState.drawing) {
+                return;
+            }
+            var featureCollection = event.getGeoJson();
+            var editSuccess = editState.executeGeometryOp(featureCollection.features[0]);
+            if (editSuccess) {
+                editState.showResult();
+            }
+            this.sendEvent(editState);
+            // Workaround: doing call sync would mess up DrawPlugin
+            setTimeout(editState.stopDrawing.bind(editState), 0);
+        },
+        'FeatureEvent': function (event) {
+            var op = event.getOperation();
+            if (op !== 'click') {
+                return;
+            }
+            var featureLayers = event.getFeatures();
+            var relevantLayers = featureLayers.filter(function (l) { return this._editsInProgress[l.layerId] }, this);
+            if (!relevantLayers.length) {
+                return;
+            }
+            var editState = this._editsInProgress[relevantLayers[0].layerId];
+            var newSelectedIndex = relevantLayers[0].geojson.features[0].properties.id;
+            if (editState && typeof newSelectedIndex === 'number') {
+                editState.selectedFeatureIndex = newSelectedIndex;
+                editState.showResult();
+                this.sendEvent(editState);
+            }
+        }
+    },
+    requestHandlers: {
+        'StartGeometryCuttingRequest': function () {
+            return Oskari.clazz.create('Oskari.mapframework.bundle.geometrycutter.StartGeometryCuttingRequestHandler', this.sandbox, this);
+        },
+        'StopGeometryCuttingRequest': function () {
+            return Oskari.clazz.create('Oskari.mapframework.bundle.geometrycutter.StopGeometryCuttingRequestHandler', this.sandbox, this);
+        }
```

Comment:
As the requesthandlers are so simple you could also just add a handleRequest() function in the instance.js and return "this" as the handler for the requests. You would need to check the request.getName() in handleRequest() to check what to do. Just a thought - the current implementation is ok too.